rbac/management/principal/cleaner.py

```diff
@@ -211,8 +225,17 @@ def process_principal_events_from_umb(bootstrap_service: Optional[TenantBootstra
         if not str(e).startswith(("Already connected", "Already subscribed")):
             raise e
 
-    while UMB_CLIENT.canRead(2):  # Check if queue is empty, two sec timeout
+    while UMB_CLIENT.canRead(15):  # Check if queue is empty, 15 sec timeout
         frame = UMB_CLIENT.receiveFrame()
-        process_umb_event(frame, UMB_CLIENT, bootstrap_service)
+        if not process_umb_event(frame, UMB_CLIENT, bootstrap_service):
+            return
```

Comment:
should we close the client before returning? UMB_CLIENT.disconnect()

Reply:
Yeah definitely good catch

Reply:
Fixed and added a test for it.